packages/block-editor/src/components/block-popover/index.js

```diff
@@ -47,8 +52,34 @@ function BlockPopover(
 		};
 	}, [ selectedElement, lastSelectedElement, __unstableRefreshSize ] );
 
+	const [ popoverAnchorRecomputeCounter, forceRecomputePopoverAnchor ] =
+		useReducer( ( s ) => s + 1, 0 );
+
+	// When blocks are moved up/down, they are animated to their new position by
+	// updating the `transform` property manually (i.e. without using CSS
+	// transitions or animations). The animation, which can also scroll the block
+	// editor, can sometimes cause the position of the Popover to get out of sync.
+	// A MutationObserver is therefore used to make sure that changes to the
+	// selectedElement's attribute (i.e. `transform`) can be tracked and used to
+	// trigger the Popover to rerender.
+	useLayoutEffect( () => {
+		if ( ! selectedElement ) {
+			return;
+		}
+
+		const observer = new window.MutationObserver(
+			forceRecomputePopoverAnchor
+		);
+		observer.observe( selectedElement, { attributes: true } );
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [ selectedElement ] );
+
 	const popoverAnchor = useMemo( () => {
 		if (
+			popoverAnchorRecomputeCounter < 0 ||
```

Comment:
Ah, I see that this wouldn't work so well if the counter looped back to zero or if it were a boolean. Though it could start out as `null` and the check could be `popoverAnchorRecomputeCounter`.

Reply:
That check is actually mostly there to make sure that every dependency that is listed for `useMemo` is actually used in the hook. Technically, the counter would always be equal or greater than 0 — so that check would always be `false`.

I've added an inline comment to explain this in https://github.com/WordPress/gutenberg/pull/44301/commits/6a9bd8c8899c643bca39082557366df01c3a0828

Reply:
I see, I completely misread `< 0` as `> 0` for some reason!